specs/design-black-box.md

```diff
@@ -0,0 +1,63 @@
+# GitOps Engine Design - Black Box
+
+## Summary
+
+During the elaboration of [White box](./design-white-box.md) option, it was discovered that some components are similar at a high-level but still have a lot of differences in 
+design and implementation. This is not surprising because code was developed by different teams and with a focus on different use-cases. Given that it would require a lot of
+effort to resolve such differences it is proposed contributing missing features of one project into an engine of another and use that engine in both projects.
+
+## Proposal
+
+It is proposed to use Argo CD application controller as the base for the GitOps engine and contribute a set of Flux features into it. There are two main reasons to try using Argo CD as
+a base:
+- Argo CD uses the _Application_ abstraction to represent the desired state and target the Kubernetes cluster. This abstraction works for both Argo CD and Flux.
+- The Argo CD controller leverages Kubernetes watch APIs instead of polling. This enables Argo CD features such as Health assessment, UI and could provide better performance to
+Flux as well.
+
+The following Flux features are missing in Argo CD:
+
+- Manifest generation using .flux.yaml files.
+- GPG commit signatures verification - an ability to verify the commit signature before pushing changes to the Kubernetes.
+- Namespace mode - an ability to control only given namespace in the target cluster. Currently, Argo CD requires read access in all namespaces.
+
+These features must be contributed to Argo CD before moving Argo CD controller into Argo Flux GitOps engine repo.
```

Comment:
Any reason why this is the case? Would it not be possible to move ArgoCD app controller into gitops-engine repo and add these features to the gitops-engine repo?

Reply:
I agree. The order does not really matter. Updated the sense.